src/commands/mod.rs

```diff
@@ -76,6 +80,108 @@ impl DaemonControl {
             .collect();
         ListCoinsResult { coins }
     }
+    /// gethistory retrieves a limited list of events which occured between two given dates.
+    pub fn gethistory(&self, start: u32, end: u32, limit: u64) -> Vec<HistoryEvent> {
+        let mut db_conn = self.db.connection();
+        let coins = db_conn.list_updated_coins(start, end, limit);
+
+        // All spend_txids event the spends out of bound,
+        // it is used for change detection when computing the deposit events.
+        let mut all_spend_txids: HashSet<bitcoin::Txid> = HashSet::with_capacity(coins.len());
+        // All the spends occuring in the bound.
+        let mut spends: HashMap<bitcoin::Txid, Vec<&Coin>> = HashMap::with_capacity(coins.len());
+
+        // Preparatory work to populate spends and all_spend_txids.
+        for coin in &coins {
+            if let Some(txid) = coin.spend_txid {
+                all_spend_txids.insert(txid);
+                if coin.spent_at.expect("Coin is spent") >= start
+                    && coin.spent_at.expect("Coin is spent") <= end
+                {
+                    if let Some(coins) = spends.get_mut(&txid) {
+                        coins.push(coin);
+                    } else {
+                        spends.insert(txid, vec![coin]);
+                    }
+                }
+            }
+        }
+
+        // Collect the received events
+        let mut events = Vec::with_capacity(coins.len());
+        for coin in coins.iter() {
+            // remove unconfirmed coin or change coin
```

Comment:
Why not keep unconfirmed coins?

Reply:
Why not listing them if they may disappear ?

Reply:
I think you meant "why listing them if they may disappear". In this case i'd say:
1. For the GUI it can be an awkward UX if you make a payment and it doesn't immediately appear as a spending in your homepage
2. It is very uncommon to actually cancel an unconfirmed payment. You might fee-bump the transaction, but seldom replace it to entirely cancel the payment. Therefore you'd just have an event that is slightly modified between two updates of the home page
3. In the unlikely case you actually canceled a payment, well the payment would just not be shown anymore? Only an event for the fees maybe.